### DISCUSSION TITLE
How to take input from external file

### DISCUSSION THREAD

**Discussion by maskmanlucifer** [cpeditor/cpeditor#981] [Q&A] [answered]
How i can take input from input.txt and can save output in output.txt ??.

**Comment by ouuan** [comment]
First of all, what's the reason that you want to do so? Need to submit the output file, need to run on multiple input files, or the input and output are large?

**Reply by maskmanlucifer** [reply]
I need to upload my output files in Hackercup and output might be quite large.

**Comment by ouuan** [comment, marked as answer]
On reading the input:

You can right-click on the input section and load from file, or load pairs of test cases from files in the "More" dropdown menu.

On saving the output:

-  You can right-click on the output section and save it.
-  You can specify an absolute path (e.g. `D:\hackercup\problem\output.txt`) instead of a relative path for `freopen`/`std::ofstream`.
-  You can right-click on the tab bar and "Open containing folder of Executable File". Then you can use `freopen`/`std::ofstream` with a path relative to that folder (e.g.`"output.txt"`). You may want to [change the executable file path](https://cpeditor.org/docs/preferences/language/#c-executable-file-path).

**Reply by maskmanlucifer** [reply]
Okk thank you
